stellar/controllers/stellar_deposit.go

```diff
@@ -0,0 +1,128 @@
+package controllers
+
+import (
+	"database/sql"
+	"net/http"
+	"strings"
+
+	"github.com/stellar/go/clients/horizonclient"
+	"github.com/stellar/go/protocols/horizon/operations"
+
+	"github.com/stellar/go/support/render/problem"
+	"github.com/stellar/starbridge/store"
+)
+
+var (
+	InvalidStellarTxHash = problem.P{
+		Type:   "invalid_stellar_tx_hash",
+		Title:  "Invalid Stellar Transaction Hash",
+		Status: http.StatusBadRequest,
+		Detail: "The transaction hash of the Stellar transaction is invalid.",
+	}
+	StellarTxHashNotFound = problem.P{
+		Type:   "stellar_tx_hash_not_found",
+		Title:  "Stellar Transaction Hash Not Found",
+		Status: http.StatusNotFound,
+		Detail: "The stellar transaction cannot be found.",
+	}
+	StellarTxNotSuccessful = problem.P{
+		Type:   "stellar_tx_not_successful",
+		Title:  "Stellar Transaction Not Successful",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar transaction was not successful.",
+	}
+	StellarTxHasMultipleOperations = problem.P{
+		Type:   "stellar_tx_has_multiple_operations",
+		Title:  "Stellar Transaction Has Multiple Operations",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar deposit transaction is invalid because it has multiple operations.",
+	}
+	StellarTxIsNotPayment = problem.P{
+		Type:   "stellar_tx_is_not_payment",
+		Title:  "Stellar Transaction Is Not Payment",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar deposit transaction is invalid because it is not a payment.",
+	}
+	StellarTxIsNotPaymentToBridge = problem.P{
+		Type:   "stellar_tx_is_not_payment_to_bridge",
+		Title:  "Stellar Transaction Is Not Payment To Bridge",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar deposit transaction is invalid because it " +
+			"is not a payment to the bridge account.",
+	}
```

Comment:
I wonder if we can merge all these errors into `stellar_tx_invalid` and put details in `Detail` field. Unless we need to be able to check the code somewhere other than frontend it would make sense.